src/Tasks.php

```diff
@@ -867,26 +883,88 @@ public function postInstall() {
    *   If the remote database was reached and downloaded, return TRUE.
    */
   private function getDatabaseOfTruth() {
-    if (!$this->databaseSourceOfTruth()) {
-      $this->say('No source database configured.');
-      $this->say('To use this command, you must return a string from the databaseSourceOfTruth() method in your project RoboFile.php.');
+    $project_properties = $this->getProjectProperties();
+    $default_database = $this->databaseSourceOfTruth();
+
+    if (file_exists('vendor/database.sql.gz')) {
+      $default_database = 'local';
+    }
+
+    $this->say('Emptying existing database.');
+    $empty_database = $this->taskExec('drush sql-drop -y @self')->dir($project_properties['web_root'])->run();
+
+    $this->say('This command populates your database from a backup .sql.gz file.');
+    $this->say('If you already have a database backup in your  vendor folder, the "local" option will be available.');
+    $this->say(' If you want to grab a more recent backup from Pantheon, type in the environment name (dev, test, live). This will be saved to your vendor folder for future re-installs.');
+    $this->say('Backups are generated on Pantheon regularly, but might be old.');
+    $this->say('If you need the very latest data from a Pantheon site, go create a new backup using either the Pantheon backend, or Terminus.');
+
+    $which_database = $this->askDefault(
+      'Which database backup should we load (local/dev/live)?', $default_database
+    );
+
+    if ($which_database !== 'local') {
+      $getDB = $this->downloadPantheonBackup($which_database);
+    }
```

Comment:
Little rework here to move the sql-drop command down to the bottom of this function. Provides a couple benefits/safety to your local db... 1. If the pantheon backup fails, you aren't left with an empty database. 2. It gives the user an opportunity to bail with `Ctrl+C` at the `Which database backup should we load (local/dev/live)?` prompt without nuking their local database:

```php
$getDB = TRUE;
if ($which_database !== 'local') {
  $getDB = $this->downloadPantheonBackup($which_database);
}

if ($getDB) {
  $this->say('Emptying existing database.');
  $empty_database = $this->taskExec('drush sql-drop -y @self')->dir($project_properties['web_root'])->run();
}
else {
  //What should we do here? Thinking we should probably throw an error and exit immediately instead of trying to import the wrong db dump...
}
```

Reply:
And given this change, we should also be explicit in the text above (I know it's already LONG 😬)... Something like:

```
$this->say('This command will drop all tables in your local database and re-populate from a backup .sql.gz file.');
```

Reply:
These are great suggestions. Thanks @wxactly!

Reply:
I made the changes listed above. Now if you do something like ask for a pantheon backup that doesn't exist, like `junk` you get the warning from the downloadPantheonBackup function:
`Failed to find a recent backup for the wfw8.junk site. Does one exist?`

AND this warning:

`Failed to download a Pantheon backup. Database was not refreshed.`